### PR TITLE
Prevent superuser from being a connection pool user

### PR DESCRIPTION
**Pull request by erthalion**
It's not supported, neither it's a best practice. Also fix potential
null pointer access.

### CODE REVIEW

File: pkg/controller/operator_config.go [hunk 1]
```diff
@@ -169,6 +169,11 @@ func (c *Controller) importConfigurationFromCRD(fromCRD *acidv1.OperatorConfigur
 		fromCRD.ConnectionPooler.User,
 		constants.ConnectionPoolerUserName)
 
+	if result.ConnectionPooler.User == result.SuperUsername {
+		msg := "Connection pool user is not allowed to be the same as super user, username: %s"
+		panic(fmt.Errorf(msg, result.ConnectionPooler.User))
+	}
```
Comment (FxKu) [inline pkg/controller/operator_config.go:175]: isn't it validated in `config.go` anyway?

**Comment by FxKu** [comment]
:+1:

File: pkg/cluster/cluster.go [hunk 2]
```diff
@@ -877,9 +877,16 @@ func (c *Cluster) initSystemUsers() {
...
 
-		username := util.Coalesce(
-			c.Spec.ConnectionPooler.User,
-			c.OpConfig.ConnectionPooler.User)
+		// Using superuser as pooler user is not a good idea. First of all it's
+		// not going to be synced correctly with the current implementation,
+		// and second it's a bad practice.
+		username := c.OpConfig.ConnectionPooler.User
+
+		if c.Spec.ConnectionPooler.User != c.OpConfig.SuperUsername {
```
Comment (sdudoladov) [inline pkg/cluster/cluster.go:885]: Does it make sense to avoid names from `protected_role_names` and `replication_username` as well ?

Comment (erthalion) [reply pkg/cluster/cluster.go]: Good point. Yes, I believe it makes sense.

**Comment by sdudoladov** [comment]
:+1:

**Comment by FxKu** [comment]
:+1:

**Pull request merged**